tests/hf/test_hartree_fock.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for for Hartree-Fock functions.
+"""
+import autograd
+import pytest
+from pennylane import numpy as np
+from pennylane.hf.hartree_fock import generate_scf, hf_energy, nuclear_energy
+from pennylane.hf.molecule import Molecule
+
+
+@pytest.mark.parametrize(
```

Comment:
Maybe you could separate the tests in different classes. It will help if we need to add more of them in the future.

Reply:
Thanks @rmoyard, yes, I will organise the tests in different classes when the unrestricted part is implemented.